services/venice-controller/src/main/java/com/linkedin/venice/controller/TerminalStateTopicCheckerForParentController.java

```diff
@@ -117,10 +118,10 @@ public void run() {
    * corresponding Venice cluster.
    */
   private Map<String, Map<String, Long>> getRelevantVeniceVersionTopics() {
-    Map<String, Long> topicRetentions = parentController.getTopicManager().getAllTopicRetentions();
+    Map<PubSubTopic, Long> topicRetentions = parentController.getTopicManager().getAllTopicRetentions();
     Map<String, Map<String, Long>> allVeniceVersionTopics = new HashMap<>();
-    for (Map.Entry<String, Long> entry: topicRetentions.entrySet()) {
-      String topic = entry.getKey();
+    for (Map.Entry<PubSubTopic, Long> entry: topicRetentions.entrySet()) {
+      String topic = entry.getKey().getName();
       try {
         if (!Version.isRealTimeTopic(topic) && Version.isVersionTopicOrStreamReprocessingTopic(topic)) {
```

Comment:
nit: Just found this kind of logic everywhere which is not part of your change.
I don't think changing them in the gigantic PR helps, but would be good to clean up and hide in under PubSubTopicType.

Reply:
Will try to do it in the future change.